From f55cca0e86f59ec11ffafd5c0503c39ca3723e2e Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Mon, 4 Feb 2019 17:32:56 +0100
Subject: [PATCH] libtest: compile lib1560.c with -fno-builtin-strcmp

... to prevent valgrind from reporting false positives on x86_64:

Conditional jump or move depends on uninitialised value(s)
   at 0x10BCAA: part2id (lib1560.c:489)
   by 0x10BCAA: updateurl (lib1560.c:521)
   by 0x10BCAA: set_parts (lib1560.c:630)
   by 0x10BCAA: test (lib1560.c:802)
   by 0x4923412: (below main) (in /usr/lib64/libc-2.28.9000.so)

Conditional jump or move depends on uninitialised value(s)
   at 0x10BCC3: part2id (lib1560.c:491)
   by 0x10BCC3: updateurl (lib1560.c:521)
   by 0x10BCC3: set_parts (lib1560.c:630)
   by 0x10BCC3: test (lib1560.c:802)
   by 0x4923412: (below main) (in /usr/lib64/libc-2.28.9000.so)
---
 tests/libtest/Makefile.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/libtest/Makefile.inc b/tests/libtest/Makefile.inc
index 080421b..ea3b806 100644
--- a/tests/libtest/Makefile.inc
+++ b/tests/libtest/Makefile.inc
@@ -531,6 +531,7 @@ lib1559_SOURCES = lib1559.c $(SUPPORTFILES) $(TESTUTIL) $(WARNLESS)
 lib1559_LDADD = $(TESTUTIL_LIBS)
 
 lib1560_SOURCES = lib1560.c $(SUPPORTFILES) $(TESTUTIL) $(WARNLESS)
+lib1560_CFLAGS = $(AM_CFLAGS) -fno-builtin-strcmp
 lib1560_LDADD = $(TESTUTIL_LIBS)
 
 lib1591_SOURCES = lib1591.c $(SUPPORTFILES) $(TESTUTIL) $(WARNLESS)
-- 
2.17.2

